intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordListener.kt

```diff
@@ -102,8 +130,21 @@ class MirrordListener : ExecutionListener {
         super.processTerminating(executorId, env, handler)
     }
 
-    private fun getRunConfigEnv(env: ExecutionEnvironment): LinkedHashMap<String, String> {
-        val envMethod = env.runProfile.javaClass.getMethod("getEnvs")
-        return envMethod.invoke(env.runProfile) as LinkedHashMap<String, String>
+    private fun getRunConfigEnv(env: ExecutionEnvironment): LinkedHashMap<String, String>? {
+        val method = when (env.runProfile::class.simpleName) {
+            "GoApplicationConfiguration" -> return null
+            else -> "getEnvs"
+        }
+        return try {
+            val envMethod = env.runProfile.javaClass.getMethod(method)
```

Comment:
So what happens here for Go? Does it always fail because the method is null?

Reply:
`"GoApplicationConfiguration" -> return null` if it is go we just return null from the function, cant find an Option type or anything along the lines. will look if there is a better way

Reply:
Ah, haven't noticed the `return`. Confusing IMO to return from inside the if clause meant to assign a value to `method`. Why not just start the function with `return if env.runProfile::class.simpleName == 'GoApplicationConfiguration`?